mne/epochs.py

```diff
@@ -1741,7 +1748,17 @@ def _get_data(
             del drop_log
 
             self._bad_dropped = True
-            logger.info("%d bad epochs dropped" % (n_events - len(good_idx)))
+            n_bads_dropped = n_events - len(good_idx)
+            logger.info(f"{n_bads_dropped} bad epochs dropped")
+
+            if n_bads_dropped == n_events:
+                logger.warning(
```

Comment:
Don't use `logger.warning` but instead `mne.utils.warn(...)`, it does some nice `verbose`-based magic

Reply:
I am probably making a mistake here, but I can't get my new test to pass. They always fail on the RuntimeWarning and with `pytest.warns(RuntimeWarning)` it fails anyway at `warn_explicit`. @larsoner Can you help me there?

Reply:
Other tests also seem to be affected if I replace `logger.warning` with `warn`